src/xdsl/dialects/arith.py

```diff
@@ -0,0 +1,215 @@
+            raise Exception("expect all input and output types to be equal")
+
+    @staticmethod
+    def get(operand1: Union[Operation, SSAValue],
+            operand2: Union[Operation, SSAValue]) -> Muli:
+        return Muli.build(operands=[operand1, operand2],
+                          result_types=[IntegerType.from_width(32)])
+
+
+@irdl_op_definition
+class Subi(Operation):
+    name: str = "arith.subi"
+    input1 = OperandDef(IntegerType)
+    input2 = OperandDef(IntegerType)
+    output = ResultDef(IntegerType)
+
+    # TODO replace with trait
+    def verify_(self) -> None:
+        if self.input1.typ != self.input2.typ or self.input2.typ != self.output.typ:
+            raise Exception("expect all input and output types to be equal")
+
+    @staticmethod
+    def get(operand1: Union[Operation, SSAValue],
+            operand2: Union[Operation, SSAValue]) -> Subi:
+        return Subi.build(operands=[operand1, operand2],
+                          result_types=[IntegerType.from_width(32)])
+
+
+@irdl_op_definition
+class FloorDiviSI(Operation):
+    name: str = "arith.floordivsi"
+    input1 = OperandDef(IntegerType)
+    input2 = OperandDef(IntegerType)
+    output = ResultDef(IntegerType)
+
+    # TODO replace with trait
+    def verify_(self) -> None:
+        if self.input1.typ != self.input2.typ or self.input2.typ != self.output.typ:
+            raise Exception("expect all input and output types to be equal")
+
+    @staticmethod
+    def get(operand1: Union[Operation, SSAValue],
+            operand2: Union[Operation, SSAValue]) -> FloordiviSigned:
+        return FloordiviSigned.build(operands=[operand1, operand2],
+                                     result_types=[IntegerType.from_width(32)])
+
+
+@irdl_op_definition
+class RemSI(Operation):
+    name: str = "arith.remsi"
+    input1 = OperandDef(IntegerType)
+    input2 = OperandDef(IntegerType)
+    output = ResultDef(IntegerType)
+
+    # TODO replace with trait
+    def verify_(self) -> None:
+        if self.input1.typ != self.input2.typ or self.input2.typ != self.output.typ:
+            raise Exception("expect all input and output types to be equal")
+
+    @staticmethod
+    def get(operand1: Union[Operation, SSAValue],
+            operand2: Union[Operation, SSAValue]) -> RemiSigned:
+        return RemiSigned.build(operands=[operand1, operand2],
+                                result_types=[IntegerType.from_width(32)])
+
+
+@irdl_op_definition
+class And(Operation):
+    name: str = "arith.and"
+    input1 = OperandDef(IntegerType)
+    input2 = OperandDef(IntegerType)
+    output = ResultDef(IntegerType)
+
+    # TODO replace with trait
+    def verify_(self) -> None:
+        if self.input1.typ != self.input2.typ or self.input2.typ != self.output.typ:
+            raise Exception("expect all input and output types to be equal")
+
+    @staticmethod
+    def get(operand1: Union[Operation, SSAValue],
```

Comment:
The MLIR and can work on different bitwidths, whereas we seem to only return I1's.